### PR TITLE
UCT/GGA: fix compilation --with-mlx5-dv=no --with-devx=no

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
fix build `--with-mlx5-dv=no --with-devx=no`

## Why ?
to fix non-default build configuration

### CODE REVIEW

**Comment by brminich** [comment]
is it possible to add some test to CI?

File: src/uct/ib/mlx5/ib_mlx5_log.c [hunk 1]
```diff
@@ -239,6 +239,7 @@ static unsigned uct_ib_mlx5_parse_dseg(void **dseg_p, unsigned op_flags,
     return ds;
 }
 
+#if HAVE_MLX5_MMO
 static void* uct_ib_mlx5_dump_dma_seg(uct_ib_mlx5_dma_seg_t *dma_seg,
```
Comment (tvegas1) [inline src/uct/ib/mlx5/ib_mlx5_log.c:243]: need to wrap `uct_ib_mlx5_dump_dma_seg` calls with `HAVE_MLX5_MMO` too, or do we have a way to put #if/#end inside `uct_ib_mlx5_dump_dma_seg`?

**Comment by evgeny-leksikov** [comment]
> is it possible to add some test to CI?

actually, `--with-devx=no` is enough to reproduce and we have such test, probably it was disabled by some reason

File: src/uct/ib/mlx5/ib_mlx5_log.c (hunk 1 above)
Comment (evgeny-leksikov) [reply src/uct/ib/mlx5/ib_mlx5_log.c]: yeah, already caught and fixed

**Comment by brminich** [comment]
> > is it possible to add some test to CI?
> 
> actually, `--with-devx=no` is enough to reproduce and we have such test, probably it was disabled by some reason

can you pls check why it did not caught the problem and fix that?

**Comment by evgeny-leksikov** [comment]
> > > is it possible to add some test to CI?
> > 
> > 
> > actually, `--with-devx=no` is enough to reproduce and we have such test, probably it was disabled by some reason
> 
> can you pls check why it did not caught the problem and fix that?

https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=90283&view=logs&j=b225495a-a138-58be-0e0c-72778e1ff6b6&t=c6c5013e-bc26-58e0-88ac-2d26030ca3ad&l=69067 the problem was caught but not handled

**Comment by evgeny-leksikov** [comment]
thanks to @tvegas1 for the fix!
let's see how it goes and how many failures we missed

**Comment by evgeny-leksikov** [comment]
link to another build fix https://github.com/openucx/ucx/pull/10235

**Comment by tvegas1** [comment]
> link to another build fix #10235

We now see expected CI failures. #10235 will fix them.

File: buildlib/tools/builds.sh [hunk 2]
```diff
@@ -461,5 +461,5 @@ do
 	[ -d "${ucx_build_dir}" ] && rm -rf ${ucx_build_dir}/*
 
 	# run the test
-	$test_name || { azure_log_error "Test failed: $test_name"; exit 1; }
+	$test_name
```
Comment (yosefe) [inline buildlib/tools/builds.sh:464]: how did it hid the build issues?

**Review by yosefe — APPROVED** [review]

File: buildlib/tools/builds.sh (hunk 2 above)
Comment (tvegas1) [reply buildlib/tools/builds.sh:464]: when using `||` or `if !` on a bash function, seems `set -e` is ignored:
```shell
#!/bin/bash -eEl

run() {
    echo options set $-
    # grep returns error $? 2 like make
    grep abc abc
    echo Should never be seen
}

echo == broken 1
run || echo FAIL 1

echo == broken 2
if ! run; then
    echo FAIL 2
fi

echo == broken 3
( run ) || echo FAIL 3

echo == broken 3
{ run; } || echo FAIL 4

echo == success
run
echo Should not be seen 2
```

```
$ ./a.sh
== broken 1
options set ehBE
grep: abc: No such file or directory
Should never be seen
== broken 2
options set ehBE
grep: abc: No such file or directory
Should never be seen
== broken 3
options set ehBE
grep: abc: No such file or directory
Should never be seen
== broken 3
options set ehBE
grep: abc: No such file or directory
Should never be seen
== success
options set ehBE
grep: abc: No such file or directory
```

**Pull request merged**